gapir/cc/resource_cache.cpp

```diff
@@ -23,51 +23,116 @@
 #include <vector>
 
 namespace gapir {
-size_t ResourceCache::prefetch(const Resource* res, size_t count,
-                               ResourceLoader* fetcher) {
-  size_t res_sum = 0;
-  std::vector<Resource> uncached;
-  uncached.reserve(count);
-  size_t alreadyCached = 0;
+
+size_t ResourceCache::setPrefetch(const Resource* resources, size_t count,
+                                  std::unique_ptr<ResourceLoader> fetcher) {
+  mResources.clear();
+  mResourceIterators.clear();
+
+  mResources.reserve(count);
+
+  for (unsigned int i = 0; i < count; ++i) {
+    mResources.push_back(resources[i]);
+    mResourceIterators[resources[i].getID()] = mResources.end() - 1;
```

Comment:
I kind of feel this is overkill: can't you have a simple map(resourceid -> resource) ?